Implement a graphql endpoint

awead opened the pull request:
Basic endpoint for querying known resource ids using Graphql. It will return all the metadata that a public user may find. Authentication is not required.

mtribone commented:
![screen shot 2017-07-11 at 9 26 01 am](https://user-images.githubusercontent.com/4163828/28070530-38b5acaa-661b-11e7-95f7-22ad47eecd0d.png)

cam156 commented on spec/controllers/catalog_controller_spec.rb:
```diff
@@ -134,13 +133,6 @@
         expect(assigns(:document_list).count).to be(1)
         expect(assigns(:document_list)[0].fetch(solr_field("resource_type"))[0]).to eql('resource_typeresource_type')
       end
-      it "finds a file by format_label" do
```
@mtribone does this test fail?  Since you are still including the facet I would expect this test to pass

mtribone replied:
@cam156 Yes this test was failing until I removed it.

mtribone replied:
https://travis-ci.org/psu-stewardship/scholarsphere/jobs/252117848#L2250

cam156 replied:
@mtribone since you kept the facet that test "Should" pass.  There may need to be a tweak to the test now that the facet is not displayed the results.

cam156 commented on spec/controllers/my/works_controller_spec.rb:
```diff
@@ -134,13 +134,6 @@
         expect(assigns(:document_list).count).to eql(1)
         expect(assigns(:document_list)[0].fetch(solr_field("resource_type"))[0]).to eql('resource_typeresource_type')
       end
-      it "finds a file by format_label" do
```
@mtribone I thought this was going to be back in.

mtribone replied:
@cam156 Oops. forgot about the works controller

awead committed:
Implement a graphql endpoint

Basic endpoint for querying known resource ids using Graphql. It will
return all the metadata that a public user may find. There is no
authentication currently.

awead committed:
Remove unused Graphql types
